compress_logs.py

```diff
@@ -0,0 +1,47 @@
+import re
+import sys
+import argparse
+import os
+
+def compress_logs(log_file, patterns, output_path):
```

Comment:
this name is confusing, I mean, this is not the real compressing, right? How about a better name? :-)